### PR TITLE
Conn 1936 - update Direct, DirectCore and AdminGui with ExchangeManger

### PR DESCRIPTION
**Pull request by trantang**
paul is optional

### CODE REVIEW

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/util/ConnectionHelper.java [hunk 1]
```diff
@@ -98,44 +92,22 @@ private boolean checkLocalHcid(String EntityHcid) {
...
-        List<BusinessEntity> businessEntities = null;
+    protected List<OrganizationType> getAllOrganizations() {
+        List<OrganizationType> organizations = null;
         try {
-            businessEntities = ConnectionManagerCache.getInstance().getAllBusinessEntities();
-        } catch (ConnectionManagerException ex) {
-            LOG.error(
-                "Exception while retrieving BusinessEntities from UDDIConnectionInfo: " + ex.getLocalizedMessage(),
-                ex);
+            organizations = WebServiceProxyHelper.getAllOrganizationsByMerge();
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/util/ConnectionHelper.java]: We don't need to merge internalExchangeInfo and exchangeInfo. Remove the merge from WebserviceProxy, ExchangeManagerHelper and all related code.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/exchangemgr/ExchangeManager.java [hunk 2]
```diff
@@ -177,4 +176,8 @@ protected String getAPI_SPEC(UDDI_SPEC_VERSION api_spec) {
     protected UDDI_SPEC_VERSION getAPI_SPEC_ENUM(String version) {
         return NhincConstants.UDDI_SPEC_VERSION.fromString(version);
     }
+
+    public static ExchangeInfoType getExchangeInfo() throws ExchangeManagerException {
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/exchangemgr/ExchangeManager.java]: No need for this method

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/exchangemgr/ExchangeManagerHelper.java [hunk 3]
```diff
@@ -161,4 +175,251 @@ public EndpointType getServiceEndpointType(OrganizationType org, String serviceN
...
+        boolean requiredElement) {
+        return getOrganizationTypeBy(findExchangeTypeBy(getExchangeTypeBy(exchangeInfo, requiredElement), exchangeName),
+            requiredElement);
+    }
+
+    public static OrganizationType findOrganizationTypeBy(List<OrganizationType> organizations, String hcid) {
+        if (CollectionUtils.isEmpty(organizations)) {
+            return null;
+        }
+        if (StringUtils.isBlank(hcid)) {
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/exchangemgr/ExchangeManagerHelper.java:211]: Please elaborate why we are returning an organization when hcid is blank

**Review by TabassumJafri — CHANGES_REQUESTED** [review]

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/exchangemgr/ExchangeManagerHelper.java (hunk 3 above)
Comment (trantang) [reply Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/exchangemgr/ExchangeManagerHelper.java:211]: hcid is now required and null is return when hcid-isBlank

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/display/FhirDisplayController.java [hunk 4]
```diff
@@ -41,6 +41,7 @@
 public class FhirDisplayController implements DisplayController {
 
     private static final Logger LOG = LoggerFactory.getLogger(FhirDisplayController.class);
+    private static final WebServiceProxyHelper HELPER = new WebServiceProxyHelper();
```
Comment (mhpnguyen) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/display/FhirDisplayController.java]: name convention: use proxyHelper

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/ConnectionManagerBean.java [hunk 5]
```diff
@@ -95,66 +96,38 @@ public void setSelectedEntityName(String selectedEntityName) {
...
 
     public String getSelectedEntityDescription() {
         String description = NULL_DISPLAY;
-
-        if (selectedEntity != null && selectedEntity.getDescription() != null
-            && !selectedEntity.getDescription().isEmpty()) {
-            description = selectedEntity.getDescription().get(0).getValue();
-        }
-
+        // OrganizationType no longer have Description
```
Comment (mhpnguyen) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/ConnectionManagerBean.java]: this method only return nulldisplay. can we just remove it then

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/FhirResourceServiceImpl.java [hunk 6]
```diff
@@ -59,6 +59,7 @@
     private static final String BINARY_DISPLAY = "Binary Resource";
 
     private static final Logger LOG = LoggerFactory.getLogger(FhirResourceServiceImpl.class);
+    private static final WebServiceProxyHelper HELPER = new WebServiceProxyHelper();
```
Comment (mhpnguyen) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/FhirResourceServiceImpl.java]: name convention

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/StatusServiceImpl.java [hunk 7]
```diff
@@ -105,47 +98,42 @@ public String getApplicationServer() {
...
-                spec.toString());
-            if (bindingTemplate != null && bindingTemplate.getAccessPoint() != null) {
-                aService.setAvailable(pingService.ping(bindingTemplate.getAccessPoint().getValue()));
-                namedServices.add(aService);
+    private static List<AvailableService> getAvailableServiceBy(EndpointType endpoint) {
+        List<AvailableService> services = new ArrayList<>();
+        if (null != endpoint) {
+            for (String serviceName : endpoint.getName()) {
+                for (EndpointConfigurationType url : getEndpointConfigurationTypeBy(endpoint)) {
+                    if (url != null) {
```
Comment (mhpnguyen) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/StatusServiceImpl.java]: url never null when going inside that loop. remove unnecessary check

**Review by mhpnguyen — CHANGES_REQUESTED** [review]

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/ConnectionManagerBean.java (hunk 5 above)
Comment (TabassumJafri) [reply Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/ConnectionManagerBean.java]: Need to decide if we want description in new exchange schema.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/exchangemgr/InternalExchangeManager.java [hunk 8]
```diff
@@ -161,12 +162,44 @@ public String getEndpointURL(String hcid, String sServiceName,
...
+            return false;
+        }
+
+        EndpointConfigurationType endpointUrl = HELPER.getEndPointConfigBasedOnSpecVersion(
+            HELPER.getServiceEndpointType(newOrganization, serviceName),
+            NhincConstants.ADAPTER_API_LEVEL.LEVEL_a0.name());
+        endpointUrl.setUrl(url);
+
+        ExchangeInfoType exchangeInfo = getExchangeInfo();
+        List<OrganizationType> organizations = HELPER.getOrganizationTypeBy(exchangeInfo);
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/exchangemgr/InternalExchangeManager.java]: Can you use List<OrganizationType> getAllOrganizations() method from AbstractExchangeManager here?

**Review by TabassumJafri — CHANGES_REQUESTED** [review]

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/display/FhirDisplayController.java (hunk 4 above)
Comment (jsmith2) [reply Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/display/FhirDisplayController.java]: agree.  HELPER is too generic of a variable name.

**Review by TabassumJafri — APPROVED** [review]

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/display/FhirDisplayController.java [hunk 9]
```diff
@@ -41,6 +41,7 @@
 public class FhirDisplayController implements DisplayController {
 
     private static final Logger LOG = LoggerFactory.getLogger(FhirDisplayController.class);
+    private static final WebServiceProxyHelper PROXY_HELPER = new WebServiceProxyHelper();
```
Comment (mhpnguyen) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/display/FhirDisplayController.java:44]: Optional: don't use upper case for variable.  Only use upper case for constant variable instead.

**Review by mhpnguyen — APPROVED** [review]

**Pull request merged**